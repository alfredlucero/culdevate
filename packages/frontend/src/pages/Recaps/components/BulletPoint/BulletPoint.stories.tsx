import React from "react";
import { storiesOf } from "@storybook/react";
import BulletPoint from "./index";

const bulletPoints = [
  "Migrated web app from on-prem nginx servers to AWS S3 and CloudFront",
  "Collaborated with other developers in creating a redesigned Email Activity",
  "Led the Anti-Spoofing project in building a new Single Sender Verification page and revised onboarding flows",
];

storiesOf("Common/BulletPoint", module).add("List of Bullet Points", () => (
  <div>
    <ul>
      {bulletPoints.map((bulletPoint, idx) => {
        return <BulletPoint bulletPoint={bulletPoint} key={idx} className="mb-2" />;
      })}
    </ul>
  </div>
));
