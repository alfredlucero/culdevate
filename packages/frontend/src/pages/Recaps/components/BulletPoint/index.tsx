import React from "react";
import cn from "classnames";
import Icon from "../../../../components/Icon";
import Text from "../../../../components/Text";
import { CommonProps } from "../../../../components/commonProps";

interface BulletPointProps extends CommonProps {
  bulletPoint: string;
}

const BulletPoint: React.FC<BulletPointProps> = ({ bulletPoint, testId = "", className = "", ...passThroughProps }) => {
  return (
    <li className={cn("flex", className)} {...(testId !== "" ? { "data-testid": testId } : {})} {...passThroughProps}>
      <Icon variant="bulletpoint" size="medium" className="mr-2 text-teal-500" />
      <Text variant="p" className="pt-1">
        {bulletPoint}
      </Text>
    </li>
  );
};

export default BulletPoint;
