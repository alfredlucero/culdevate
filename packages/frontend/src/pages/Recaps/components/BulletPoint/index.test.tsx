import React from "react";
import { render } from "@testing-library/react";
import BulletPoint from "./index";

describe("<BulletPoint />", () => {
  const bulletPointTestId = "bulletPointTestId";

  test("should render without error", () => {
    const { container } = render(
      <BulletPoint bulletPoint="Some bullet point!" testId={bulletPointTestId} className="extra-bulletpoint-class" />,
    );

    expect(container).toMatchSnapshot();
  });
});
