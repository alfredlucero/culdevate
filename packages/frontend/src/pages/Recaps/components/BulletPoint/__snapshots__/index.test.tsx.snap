// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BulletPoint /> should render without error 1`] = `
<div>
  <li
    class="flex extra-bulletpoint-class"
    data-testid="bulletPointTestId"
  >
    <div
      class="font-sans text-gray-800 text-2xl mr-2 text-teal-500"
      id=""
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-chevron-circle-right fa-w-16 "
        data-icon="chevron-circle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
          fill="currentColor"
        />
      </svg>
    </div>
    <p
      class="font-sans text-gray-800 text-base pt-1"
    >
      Some bullet point!
    </p>
  </li>
</div>
`;
